addonscript/script.py

```diff
@@ -0,0 +1,73 @@
+import aiohttp
+import asyncio
+import os
+
+import scriptworker.client
+
+from addonscript.api import do_upload, get_upload_status, get_signed_xpi
+from addonscript.task import build_filelist
+from addonscript.xpi import get_langpack_info
+
+
+def _craft_aiohttp_connector(context):
+    return aiohttp.TCPConnector()
+
+
+def get_default_config(base_dir=None):
+    base_dir = base_dir or os.path.dirname(os.getcwd())
+    default_config = {
+        'work_dir': os.path.join(base_dir, 'work_dir'),
+        'schema_file': os.path.join(os.path.dirname(__file__), 'data', 'addonscript_task_schema.json'),
+    }
+    return default_config
+
+
+async def sign_addon(context, locale):
+    # Retry Me
+    if 1:
+        upload_data = await do_upload(context, locale)
+        pk = upload_data['pk']
+    # Retry me, instead of sleeping
+    if 1:
+        await asyncio.sleep(60*6)  # wait 6 minutes
```

Comment:
We may want this configurable, depending on what it's for. Why the wait? `do_upload` has finished by this point. I'm guessing there's a lag after the upload finishes, but before `get_upload_status` gets info?

Reply:
This was mostly a "do I implement retry first", the signing can happen between 5 and 10 minutes after upload, but usually sooner.